scheduler/src/cook/mesos/task.clj

```diff
@@ -41,8 +67,8 @@
         container (util/job-ent->container db job-ent)
         ;; If the custom-executor attr isn't set, we default to using a custom
         ;; executor in order to support jobs submitted before we added this field
-        custom-executor? (:job/custom-executor job-ent true)
-        cook-executor? (and (not custom-executor?) (seq executor-config))
+        custom-executor? (use-custom-executor? job-ent)
+        cook-executor? (use-cook-executor? job-ent executor-config)
```

Comment:
We should note which executor the instance used if it can be different between instances

Reply:
Added the executor to the job instance entity in `cook.mesos.scheduler/matches->task-txns`